src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SingleField.jsx

```diff
@@ -1,46 +1,59 @@
-import React, {useState} from 'react';
+import React, {useState, useEffect, useReducer} from 'react';
 import * as PropTypes from 'prop-types';
 import {compose} from '~/utils';
 import {connect, FastField} from 'formik';
 import {FieldPropTypes} from '~/FormDefinitions';
 
+const reducer = (state, action) => {
+    switch (action.type) {
+        case 'set':
+            return action.data;
+        case 'unMount':
+            return action.onChange(state, undefined);
+        default:
+            throw new Error();
+    }
+};
+
```

Comment:
we can move this reducer to the field level, to be used by the single and multiple and to avoid to have two fragments of code doing the same thing.

Reply:
yes, it is a possible improvement, but it works in current state

Reply:
hahahaha this is the TOP comment :D